src/main.js

```diff
@@ -74,133 +85,205 @@ const startServer = (configPath, config) => {
   });
 };
 
-const startWithMBTiles = (mbtilesFile) => {
-  console.log(`[INFO] Automatically creating config file for ${mbtilesFile}`);
+const startWithinputFile = async (inputFile) => {
+  console.log(`[INFO] Automatically creating config file for ${inputFile}`);
   console.log(`[INFO] Only a basic preview style will be used.`);
   console.log(
     `[INFO] See documentation to learn how to create config.json file.`,
   );
 
-  mbtilesFile = path.resolve(process.cwd(), mbtilesFile);
+  let inputFilePath;
+  if (isValidHttpUrl(inputFile)) {
+    inputFilePath = process.cwd();
+  } else {
+    inputFile = path.resolve(process.cwd(), inputFile);
+    inputFilePath = path.dirname(inputFile);
 
-  const mbtilesStats = fs.statSync(mbtilesFile);
-  if (!mbtilesStats.isFile() || mbtilesStats.size === 0) {
-    console.log(`ERROR: Not valid MBTiles file: ${mbtilesFile}`);
-    process.exit(1);
-  }
-  const instance = new MBTiles(mbtilesFile + '?mode=ro', (err) => {
-    if (err) {
-      console.log('ERROR: Unable to open MBTiles.');
-      console.log(`Make sure ${path.basename(mbtilesFile)} is valid MBTiles.`);
+    const inputFileStats = fs.statSync(inputFile);
+    if (!inputFileStats.isFile() || inputFileStats.size === 0) {
+      console.log(`ERROR: Not a valid input file: `);
       process.exit(1);
     }
+  }
 
-    instance.getInfo((err, info) => {
-      if (err || !info) {
-        console.log('ERROR: Metadata missing in the MBTiles.');
-        console.log(
-          `Make sure ${path.basename(mbtilesFile)} is valid MBTiles.`,
-        );
-        process.exit(1);
-      }
-      const bounds = info.bounds;
+  const styleDir = path.resolve(
+    __dirname,
+    '../node_modules/tileserver-gl-styles/',
+  );
 
-      const styleDir = path.resolve(
-        __dirname,
-        '../node_modules/tileserver-gl-styles/',
-      );
+  const config = {
+    options: {
+      paths: {
+        root: styleDir,
+        fonts: 'fonts',
+        styles: 'styles',
+        mbtiles: inputFilePath,
+        pmtiles: inputFilePath,
+      },
+    },
+    styles: {},
+    data: {},
+  };
 
-      const config = {
-        options: {
-          paths: {
-            root: styleDir,
-            fonts: 'fonts',
-            styles: 'styles',
-            mbtiles: path.dirname(mbtilesFile),
-          },
-        },
-        styles: {},
-        data: {},
-      };
-
-      if (
-        info.format === 'pbf' &&
-        info.name.toLowerCase().indexOf('openmaptiles') > -1
-      ) {
+  const extension = inputFile.split('.').pop().toLowerCase();
+  if (extension === 'pmtiles') {
+    let FileOpenInfo = PMtilesOpen(inputFile);
+    const metadata = await GetPMtilesInfo(FileOpenInfo.pmtiles);
+    if (FileOpenInfo.fd !== undefined) {
+      PMtilesClose(FileOpenInfo.fd);
+    }
+    FileOpenInfo = null;
```

Comment:
I've removed PMtilesClose for now, since I only could figure out how to use it here. At some point if we start removing sources more data dynamically I will revisit that.